### PR TITLE
API-560 get/upload/download an asset reference file

### PR DESCRIPTION
**Pull request by LaurentPetard**
I choose to handle the upload response errors directly in `AssetReferenceFileApi` because I think it's too specific to isolate it in a service is worth it.

### CODE REVIEW

**Review by damien-carcel — APPROVED** [review]

**Review by momoss — APPROVED** [review]

File: src/Api/AssetReferenceFileApi.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    public function __construct(ResourceClientInterface $resourceClient, FileSystemInterface $fileSystem)
+    {
+        $this->resourceClient = $resourceClient;
+        $this->fileSystem = $fileSystem;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFromLocalizableAsset($assetCode, $localeCode)
```
Comment (ahocquard) [inline src/Api/AssetReferenceFileApi.php:42]: `getLocalizableAsset` / `getNotLocalizableAsset` ?

File: src/Api/AssetReferenceFileApi.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+     */
+    private function get($assetCode, $localeCode)
+    {
+        return $this->resourceClient->getResource(static::ASSET_REFERENCE_FILE_URI, [$assetCode, $localeCode]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function uploadForLocalizableAsset($referenceFile, $assetCode, $localeCode)
```
Comment (ahocquard) [inline src/Api/AssetReferenceFileApi.php:69]: `uploadLocalizableAsset` /  `uploadNotLocalizableAsset`

File: src/Api/AssetReferenceFileApi.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+     */
+    public function uploadForNotLocalizableAsset($referenceFile, $assetCode)
+    {
+        return $this->upload($referenceFile, $assetCode, 'no-locale');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function downloadFromLocalizableAsset($assetCode, $localeCode)
```
Comment (ahocquard) [inline src/Api/AssetReferenceFileApi.php:85]: `downloadLocalizableAsset` / `downloadNotLocalizableAsset`

File: src/Api/AssetReferenceFileApi.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    {
+        return $this->upload($referenceFile, $assetCode, $localeCode);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function uploadForNotLocalizableAsset($referenceFile, $assetCode)
+    {
+        return $this->upload($referenceFile, $assetCode, 'no-locale');
```
Comment (ahocquard) [inline src/Api/AssetReferenceFileApi.php]: `no-locale` in a const
 `NOT_LOCALIZABLE_ASSET_LOCALE` for example

**Review by ahocquard — APPROVED** [review]

File: src/Api/AssetReferenceFileApi.php (hunk 1 above)
Comment (ahocquard) [reply src/Api/AssetReferenceFileApi.php:42]: Seen IRL: we get a reference file FROM an asset. Forget my comments about naming.

**Pull request merged**